apis/core/v1beta1/function_webhook.go

```diff
@@ -38,4 +43,537 @@ var _ webhook.Defaulter = &Function{}
+				return field.NotSupported(field.NewPath("spec", "serving", "scaleOptions",
+					"keda", "scaleObject", "advanced", "horizontalPodAutoscalerConfig", "behavior", "scaleDown", "selectPolicy"),
+					scaleDown.SelectPolicy,
+					[]string{"Max", "Min", "Disabled"})
+			}
+			for index, policy := range scaleDown.Policies {
+				if !containsString([]string{"Pods", "Percent"}, string(policy.Type)) {
+					return field.NotSupported(field.NewPath("spec", "serving", "scaleOptions",
+						"keda", "scaleObject", "advanced", "horizontalPodAutoscalerConfig", "behavior", "scaleDown",
+						"policies", fmt.Sprintf("[%d]", index), "type"),
+						policy.Type,
+						[]string{"Pods", "Percent"})
+				}
+				if policy.PeriodSeconds < 0 {
+					return field.Invalid(field.NewPath("spec", "serving", "scaleOptions",
+						"keda", "scaleObject", "advanced", "horizontalPodAutoscalerConfig", "behavior", "scaleDown",
+						"policies", fmt.Sprintf("[%d]", index), "periodSeconds"),
+						policy.PeriodSeconds,
+						"must be greater than 0")
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Function) ValidateKedaScaledJobScalingStrategy() error {
+	strategy := r.Spec.Serving.ScaleOptions.Keda.ScaledJob.ScalingStrategy
+	if strategy.Strategy != "" {
+		if !containsString([]string{"default", "custom", "accurate"}, strategy.Strategy) {
+			return field.NotSupported(field.NewPath("spec", "serving", "scaleOptions",
+				"keda", "scaleJob", "scalingStrategy", "strategy"),
+				strategy.Strategy,
+				[]string{"default", "custom", "accurate"})
+		}
+		if strategy.Strategy == "custom" && strategy.CustomScalingQueueLengthDeduction == nil {
+			return field.Required(field.NewPath("spec", "serving", "scaleOptions",
+				"keda", "scaleJob", "scalingStrategy", "customScalingQueueLengthDeduction"),
+				"must be specified when `strategy.Strategy` is custom")
+		}
+		if strategy.Strategy == "custom" && strategy.CustomScalingRunningJobPercentage == "" {
+			return field.Required(field.NewPath("spec", "serving", "scaleOptions",
+				"keda", "scaleJob", "scalingStrategy", "customScalingRunningJobPercentage"),
+				"must be specified when `strategy.Strategy` is custom")
+		}
+
+	}
+	if strategy.CustomScalingQueueLengthDeduction != nil && *strategy.CustomScalingQueueLengthDeduction < 0 {
+		return field.Invalid(field.NewPath("spec", "serving", "scaleOptions",
+			"keda", "scaleJob", "scalingStrategy", "customScalingQueueLengthDeduction"),
+			strategy.CustomScalingQueueLengthDeduction,
+			"cannot be less than 0")
+	}
+	if strategy.CustomScalingRunningJobPercentage != "" {
+		reg := regexp.MustCompile(`^([0-9.]+)[ ]*%$`)
+		if !reg.MatchString(strategy.CustomScalingRunningJobPercentage) {
+			return field.Invalid(field.NewPath("spec", "serving", "scaleOptions",
+				"keda", "scaleJob", "scalingStrategy", "customScalingRunningJobPercentage"),
+				strategy.CustomScalingRunningJobPercentage,
+				"cannot be less than 0")
```

Comment:
Would it be reasonable to prompt "is not an invalid percentage value" or smth like that?